src/Automation/Scanner.cs

```diff
@@ -24,7 +31,10 @@ internal Scanner(Config config)
         /// <returns></returns>
         public ScanResults Scan()
         {
-            throw new NotImplementedException();
+            lock (LockObject)
```

Comment:
>LockObject [](start = 18, length = 10)

SnapshotCommand.Execute calls ExecutionWrapper, which already provides a lock. Can we pick just one of the two?

Reply:
Yes. I will take this out in an upcoming commit.